src/server/pipeline.py

```diff
@@ -217,12 +218,11 @@ async def perform_handshake(
         if srwt_aiter.is_stopped():
             raise Exception("No longer accepting handshakes, closing.")
 
-        if not global_connections.add(connection):
-            raise ProtocolError(Err.DUPLICATE_CONNECTION, [False])
+        # Tries adding the connection, and raises an exception on failure.
```

Comment:
Why was this change made?

Reply:
based on Richard's previous review, to keep pipeline code as light as possible.